tools/rust_api/src/database.rs

```diff
@@ -79,10 +79,9 @@ impl Database {
     /// * `enable_compression`: When true, table data will be compressed if possible
     /// * `read_only`: When true, the database will be opened in READ_ONLY mode, otherwise in READ_WRITE mode.
     pub fn new<P: AsRef<Path>>(path: P, config: SystemConfig) -> Result<Self, Error> {
-        let_cxx_string!(path = path.as_ref().display().to_string());
         Ok(Database {
             db: UnsafeCell::new(ffi::new_database(
-                &path,
+                ffi::StringView::new(&path.as_ref().display().to_string()),
```

Comment:
I feel like this doesn't make sense. Shouldn't we pass the path unmodified to kuzu? I get that this won't matter in the majority of cases (invalid utf-8 is sadness), but since C++ is going to interpret it as bytes anyway, maybe we should keep the bytes as is.

Reply:
A `Path` in rust is "a thin wrapper over [OsString](https://doc.rust-lang.org/std/ffi/struct.OsString.html)", which, for example, on windows is a utf-16 string, not a utf-8 string, so it has to be converted if we want to treat it as utf-8. 
C++ has similar inconsistencies with `std::filesystem::path` if I remember correctly, though I seem to recall that the interfaces change slightly depending on the `std::filesystem::path` implementation, where Rust's `std::path::Path` is handled the same regardless of platform (I think it's just that on Linux/Unix std::filesystem::path can be implicitly converted to std::string, while on Windows you have to explicitly call `.string()`).

Also note that nothing's really changed in this regard. We were already calling `path.as_ref().displa().to_string()`, the only difference is that we pass it as a StringView instead converting it to a CxxString (which required two copies instead of one, as it had to be converted from OsString, to String, to CxxString).